### DISCUSSION TITLE
Технология виртуализации

### DISCUSSION THREAD

**Discussion by alglyzin** [mikopbx/Core#51] [Q&A] [answered]
Добрый день. Подскажите, поддерживается ли установка OS на машинку с OpenVZ ?

**Comment by jorikfon** [comment]
Нет, сейчас это спарка из собственного Linux + Asterisk + Web админка.
Возможна только установка на полную виртуализацию.
Точно работает в Hyper-V, vSphere облаках Яндекса и Mail.ru а также в десктопных VirtualBox и VMware Desktop.

На будущий год подумаем как отделить АТС от системы, и надо ли это делать. По крайней мере хочет в Azure, Amazon AWS и Google Cloud, а там как раз надо с их ядрами работать.

**Comment by plastilincheg** [comment]
Здравствуйте. Встроен ли в систему qemu-guest-agent для развертывания на базе KVM, в частности Proxmox

**Comment by boffart** [comment]
> Здравствуйте. Встроен ли в систему qemu-guest-agent для развертывания на базе KVM, в частности Proxmox

Нет.

**Comment by jorikfon** [comment]
Мы думаем над поддержкой разной виртуализации в будущих релизах точно реализуем.

**Comment by plastilincheg** [comment]
К сожалению банально apt install qemu-guest-agent и yum install qemu-guest-agent  - не прокатило. Что внутри за Linux? Хотя машина успешно запустилась используя драйвера VIRTIO. Также для виртуализации  машинах с несколькими ЦП важна поддержка NUMA внутри гостевой ОС. Это реализовано? Также машина не реагирует на команды ACPI. Команда Shutdown отправленная с хоста не может отключить питание.

![mikopbx_kvm](https://user-images.githubusercontent.com/9542165/94015864-17006b00-fdb6-11ea-87e8-754f130a1dcd.png)

**Comment by boffart** [comment]
ядро собрано со следующими опциями

```
CONFIG_ARCH_SUPPORTS_NUMA_BALANCING=y
# CONFIG_NUMA_BALANCING is not set
CONFIG_NUMA=y
CONFIG_AMD_NUMA=y
CONFIG_X86_64_ACPI_NUMA=y
# CONFIG_NUMA_EMU is not set
CONFIG_USE_PERCPU_NUMA_NODE_ID=y
CONFIG_ACPI_NUMA=y
# CONFIG_VIRTIO_VSOCKETS is not set
CONFIG_VIRTIO_VSOCKETS_COMMON=y
CONFIG_NET_9P_VIRTIO=y
CONFIG_VIRTIO_BLK=y
CONFIG_SCSI_VIRTIO=y
CONFIG_VIRTIO_NET=y
CONFIG_CAIF_VIRTIO=y
# CONFIG_VIRTIO_CONSOLE is not set
CONFIG_HW_RANDOM_VIRTIO=y
# CONFIG_DRM_VIRTIO_GPU is not set
CONFIG_VIRTIO=y
CONFIG_VIRTIO_PCI=y
CONFIG_VIRTIO_PCI_LEGACY=y
CONFIG_VIRTIO_BALLOON=y
CONFIG_VIRTIO_INPUT=y
CONFIG_VIRTIO_MMIO=y
CONFIG_VIRTIO_MMIO_CMDLINE_DEVICES=y
CONFIG_CRYPTO_DEV_VIRTIO=y

```
apt не прокатит, это не debian. 
linux собран согласно рекомендациям Linux Form Scratch.
Для сборки используется https://t2sde.org.

**Comment by plastilincheg** [comment]
Ну вроде все есть, вот гостевых утилит не хватает.

**Comment by irq15** [comment, marked as answer]
> 
> 
> К сожалению банально apt install qemu-guest-agent и yum install qemu-guest-agent - не прокатило. Что внутри за Linux? Хотя машина успешно запустилась используя драйвера VIRTIO. Также для виртуализации машинах с несколькими ЦП важна поддержка NUMA внутри гостевой ОС. Это реализовано? Также машина не реагирует на команды ACPI. Команда Shutdown отправленная с хоста не может отключить питание.
> 


немного не понял что вам не хватает, у меня мико на пве прекрасно работает и выключается и бекапится.
![изображение](https://user-images.githubusercontent.com/44310892/105844031-5fe15e00-5ffa-11eb-82dd-e08e72c17004.png)

**Comment by matveevandrey** [comment]
+1 к включению qemu-guest-agent

**Reply by boffart** [reply]
собранный пакет qemu-ga весит 246Мб
такой объем не допустим


еще не разобрался как это все дело работает. 
как я понимаю, хост машина отправляет агенту запросы, агент собирает информацию / выполняет какие то действия по настройке машины. 

наша ос "не стандартная", даже команда для ребута специфичная 
возможности кастомизации не вижу

мб кто документацию скинет, то, что нашел пока не радует.

**Reply by plastilincheg** [reply]
Гостевые утилиты "желательны" для корректной работы ВМ и общения с хостом. Например правильная работа с ОЗУ, питанием, информации о сети и тд.

https://wiki.qemu.org/Features/GuestAgent
https://wiki.qemu.org/Documentation/QMP

**Reply by boffart** [reply]
с этим ознакомился, но пока яснее не стало. 
будем изучать дальше...
